be/user.go

```diff
@@ -17,6 +17,7 @@ type UserArgs struct {
 
 type UserResp struct {
 	Team TeamColor `json:"team"` // Blue or Green
+	DupAvatar bool `json:"dup_avatar"`
```

Comment:
Technically, non OK status code invalidates the returned data.
And ok status means everything is fine.
How about using "409 Conflict" http code instead of hiding an error in response?

Reply:
That's a problem, I have to explicitly send what happened in the JSON response.
The status 409 doesn't necessarily mean duplicated avatar, the app cannot make a proper conclusion based on it. It needs an explicit flag in the response to understand what has happened on the server.
So I'd prefer to return a normal response with details inside.

Reply:
Technically we define http code for the specific request, so 409 for user call will uniquely identify collision on avatar. Also, technically, what I wrote is true. Althought it looks like some big players do return extra info related to an error in case of an error http code: https://www.ibm.com/docs/en/ibm-mq/9.3?topic=api-rest-error-handling
So, how about you return 409 instead of 500 in this case, but also your boolean field?
Keep in mind, technically, team field should be considered invalid in case of this error.